cylc/flow/data_store_mgr.py

```diff
@@ -2522,6 +2526,35 @@ def delta_task_xtrigger(self, sig, satisfied):
             xtrigger.time = update_time
             self.updates_pending = True
 
+    def delta_from_task_proxy(self, itask: TaskProxy) -> None:
+        """Create delta from existing pool task proxy.
+
+        Args:
+            itask (cylc.flow.task_proxy.TaskProxy):
+                Update task-node from corresponding task proxy
+                objects from the workflow task pool.
+
+        """
+        tproxy: Optional[PbTaskProxy]
+        tp_id, tproxy = self.store_node_fetcher(itask.tokens)
+        if not tproxy:
+            return
+        update_time = time()
+        tp_delta = self.updated[TASK_PROXIES].setdefault(
+            tp_id, PbTaskProxy(id=tp_id))
+        tp_delta.stamp = f'{tp_id}@{update_time}'
+        tp_existing = PbTaskProxy(id=tp_id)
+
+        self._process_internal_task_proxy(itask, tp_existing)
+        statics = {'id'}
+        for field, _ in tp_existing.ListFields():
+            old_value = getattr(tp_delta, field.name, None)
+            if old_value and field.name not in statics:
+                tp_delta.ClearField(field.name)
+        tp_delta.MergeFrom(tp_existing)
+
```

Comment:
I'm not convinced this is doing anything, actually

Reply:
We have in the general delta application..
Some protobuf fields are additive .. i.e. if the two messages contain lists, then the list would extend on `MergeFrom`, this can be a problem if we don't want duplicates in the list (i.e. we just want to replace one list with the new).. So we Clear them first.

Reply:
Is there a reason this couldn't simply be
```diff
-tp_existing = PbTaskProxy(id=tp_id)
-
-self._process_internal_task_proxy(itask, tp_existing)
+self._process_internal_task_proxy(itask, tp_delta)
-statics = {'id'}
-for field, _ in tp_existing.ListFields():
-    old_value = getattr(tp_delta, field.name, None)
-    if old_value and field.name not in statics:
-        tp_delta.ClearField(field.name)
-tp_delta.MergeFrom(tp_existing)
```

Reply:
Let me see if `_process_internal_task_proxy` wipes out the fields it's writing to .. If so, then yes, otherwise I already explained:
> Some protobuf fields are additive .. i.e. if the two messages contain lists, then the list would extend on MergeFrom, this can be a problem if we don't want duplicates in the list (i.e. we just want to replace one list with the new).. So we Clear them first.

Reply:
I had added that block in the last commit.. However, it may have been a precautionary move.

I've reviewed the code, and the protobuf fields .. The one field that would have been effected is already cleared in `_process_internal_task_proxy`, the other mappings and fields should be ok. So, I've removed that block.

Let's see how my review and removal goes..